[DOCS] Move `store` utils to individual reference pages (#483)
solidjs/solid-docs-next: src/routes/reference/store-utilities/modify-mutable.mdx

---
title: modifyMutable
---

```tsx
function modifyMutable<T>(mutable: T, modifier: (state: T) => T): void;
```

This helper function simplifies making multiple changes to a mutable Store (as returned by [createMutable](#createMutable)) in a single [batch](/reference/reactive-utilities/batch), so that dependant computations update just once instead of once per update.
The first argument is the mutable Store to modify, and the second argument is a Store modifier such as those returned by [reconcile]('#reconcile') or [produce](#produce).

<Callout type="caution">

> 	If you pass in your own modifier function, beware that its argument is an
> 	unwrapped version of the Store.

Reviewer: ```suggestion
	When passing in your own modifier function, it's important to be aware that its argument is an unwrapped version of the store.
```

</Callout>

> For example, suppose we have a UI depending on multiple fields of a mutable:

Reviewer: ```suggestion
For example, if the UI depends on multiple fields of a mutable:
```

```tsx
import { createMutable } from "solid-js/store";

const state = createMutable({
	user: {
		firstName: "John",
		lastName: "Smith",
	},
});

<h1>Hello {state.user.firstName + " " + state.user.lastName}</h1>;
```

Modifying n fields in sequence will cause the UI to update n times:

```tsx
state.user.firstName = "Jane";
state.user.lastName = "Doe";
```

> To trigger just a single update, we could modify the fields in a `batch`:

Reviewer: ```suggestion
To trigger just a single update, the fields can be modified using a `batch`:
```

```tsx
import { batch } from "solid-js";

batch(() => {
	state.user.firstName = "Jane";
	state.user.lastName = "Doe";
});
```

> `modifyMutable` combined with reconcile or produce provides two alternate ways to do similar things:

Reviewer: ```suggestion
`modifyMutable` combined with `reconcile` or `produce` provides two alternate ways to do similar things:
```
